index.html

```diff
@@ -0,0 +1,1209 @@
+
+                  <p>When a request requires an access mode (<code>acl:mode</code>) which is a limitation of another access mode, then access is granted if either mode is allowed by an Authorization. For example, when a request requires <code>acl:Append</code>, then access will be granted to agents having <code>acl:Write</code>.</p>
+
+                  <p>The presence of the <code>acl:origin</code> property and its value is taken into account in the evaluation only when the HTTP request includes the <code>Origin</code> header.</p>
+
+                  <section id="reading-writing-resources" inlist="" rel="schema:hasPart" resource="#reading-writing-resources">
+                    <h4 property="schema:name">Reading and Writing Resources</h4>
+                    <div datatype="rdf:HTML" property="schema:description">
+                      <p>Working alongside <cite><a href="#http-interaction" rel="rdfs:seeAlso">HTTP Interactions</a></cite>, WAC enables a standard set of operations which can be applied to different resource types for a given request based on the set of access modes which control those operations:</p>
+
+                      <p>Generally: read operations attempt to confirm the existence of a resource or to view the contents of a resource; write operations attempt to create, delete, or modify resources; append operations attempt to create resources or add information to existing resources; and control operations attempt to view, create, delete, or modify ACL resources.</p>
+
+                      <p>As container resources and member resources are hierarchically organised, requests to perform operations on resources are in the context of the applicable container (<cite><a href="#effective-acl-resource" rel="rdfs:seeAlso">Effective ACL Resource</a></cite>).</p>
+
+                      <p>When an operation requests to read a resource, the server MUST match an Authorization allowing the <code>acl:Read</code> access privilege on the resource.</p>
+
+                      <p>When an operation requests to create a resource as a member of a container resource, the server MUST match an Authorization allowing the <code>acl:Append</code> or <code>acl:Write</code> access privilege on the container for new members.</p>
+
+                      <p>When an operation requests to update a resource, the server MUST match an Authorization allowing the <code>acl:Append</code> or <code>acl:Write</code> access privilege on the resource.</p>
+
+                      <p>When an operation requests to delete a resource, the server MUST match Authorizations allowing the <code>acl:Write</code> access privilege on the resource and the containing container.</p>
+
```

Comment:
If there is a `mkdir -p` behaviour, make sure all entailing container creations are also allowed. Because if I only have default Write on / then I can create /foo/bar.txt if /foo/ exists, but not if the creation of /foo/ is a result of my operation.

Reply:
Not sure what needs to be clarified (noted or specified) but good to acknowledge this, thanks!

I think it is generally covered by the requirement in "to create a resource as a member of a container resource" in junction with a protocol that specifies intermediate container creation.

For now I've added a note: https://github.com/solid/web-access-control-spec/pull/83/commits/54118e7791fdbda300cedaf7b5dcd0eb8995fc94 . How's that? Intermediate container creation or recursive deletion is not required by WAC (as of this writing, as far as I know...)

Reply:
Right, yes! Please mark as resolved.